net/dcwifi/dcwapd/Makefile

```diff
@@ -0,0 +1,99 @@
+#
+# Copyright (C) 2019 EWSI
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+
+include $(TOPDIR)/rules.mk
+
+# For LINUX_DIR
+include $(INCLUDE_DIR)/kernel.mk
+
+PKG_NAME:=dcwapd
+PKG_VERSION:=1.0.0
+PKG_RELEASE:=1
+PKG_LICENSE:=Apache-2.0
+PKG_LICENSE_FILES:=COPYING
+
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
+PKG_SOURCE_URL:=https://codeload.github.com/ewsi/$(PKG_NAME)/tar.gz/v$(PKG_VERSION)?
+PKG_SOURCE_SUBDIR:=$(PKG_NAME)-$(PKG_VERSION)
+PKG_HASH:=750a08abccd88d9aeda942307f76ce5711181c06f9f3e8fded5cb5ce42bac323
+PKG_MAINTAINER:=Carey Sonsino <careys@edgewaterwireless.com>
+
+PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_SOURCE_SUBDIR)
+# Note - we use the static directory, but we patch to use shared libs
+BUILD_VERSION=linuxjsonstatic
+PKG_BUILD_BIN_DIR=$(PKG_BUILD_DIR)/dcwapd.$(BUILD_VERSION)
+
+PKG_FIXUP:=autoreconf
+
+include $(INCLUDE_DIR)/package.mk
+
+define Package/dcwapd
+  SECTION:=net
+  CATEGORY:=Network
+  SUBMENU:=Routing and Redirection
+  TITLE:=Dual-Channel WiFi AP daemon
+  URL:=https://www.edgewaterwireless.com
+  DEPENDS:= +libstdcpp +kmod-macremapper +libdcwsocket +libdcwproto +mrmctl +libuci
```

Comment:
> Can this be compiled with uClibc++ instead?

Just curious: why ?

Reply:
When I use uClibc++, I lose the ability to get verbose output from the `make` command (I cannot use the `V=s` in `make -j1 V=s package/dcwapd/compile`).  I'm getting:
`openwrt/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/bin/g++-uc: 22: [: Illegal number: s`

Line 22 of `g++-uc` is attempting to treat `$V` as a number, and I'm clearly setting it to `s`.

Is this normal behavior?

Reply:
When I updated, uClibc++, I was aware of the error but ignored it since make V=s is only for developers and not too typical. I should have fixed it.

That being said, why would it remove make output?

Reply:
Two problems. The actual error is ```mips-openwrt-linux-musl-g++: error: 0.1": No such file or directory```

Second, why is g++ being used?

Reply:
I got further into the compile. Encountered this error:

controller.cxx: In member function 'void dcw::Controller::OnStationUnjoin(const dcw::MacAddress&, const dcw::Message&)':
controller.cxx:205:24: error: assignment of member 'std::pair<dcw::MacAddress, const dcw::BasicChannel*>::second' in read-only object
     dcmEntry->second = NULL;
                        ^~~~
controller.cxx: In member function 'void dcw::Controller::OnStationAck(const dcw::MacAddress&, const dcw::Message&)':
controller.cxx:252:32: error: binding reference of type 'dcw::Controller::ClientState&' to 'const dcw::Controller::ClientState' discards qualifiers
   ClientState& state = client->second;

Reply:
Oh it's just a const error.

Anyway, the reason it's failing is because of https://github.com/openwrt/openwrt/blob/master/package/libs/uclibc%2B%2B/patches/002-quote-fix.patch

Remove that and it will compile.

I added that patch as one package was not compiling without it. It turns out, that package was actually totally broken to begin with. It has since been fixed (error exposed with PKG_ASLR_PIE).

I will post a patch for uClibc++ on the mailing list.

Reply:
It seems rapidjson requires C++11. This means it's no go as uClibc++ is a C++98 library. For the rest of the errors, here's a patch:

https://gist.github.com/neheb/8b6223d52437b52a29e360220946deed

Reply:
Actually it's doable. RAPIDJSON_HAS_CXX11_RVALUE_REFS just has to be defined to 0.

Reply:
Awesome, thanks for the assistance and patch!